via/static/scripts/video_player/components/ToastMessages.tsx

```diff
@@ -0,0 +1,190 @@
+    // Ignore non-relevant animations:
+    // * Happening on children elements
+    // * Animations which are not relevant for toast messages getting "in" or "out"
+    if (
+      e.target !== containerRef.current ||
+      !['slide-in-from-right', 'fade-in', 'fade-out'].includes(e.animationName)
+    ) {
+      // We cannot test this scenario. It will be more relevant when/if this
+      // component is extracted at some point
+      /* istanbul ignore next */
+      return;
+    }
+
+    onTransitionEnd?.(direction);
+  };
+
+  return (
+    <div
+      onAnimationEnd={handleAnimation}
+      ref={containerRef}
+      className={classnames('relative w-full container', {
+        // Only ever slide-from-right if motion-safe is preferred
+        'lg:motion-safe:animate-slide-in-from-right animate-fade-in':
+          !isDismissed,
+        // Only ever fade-in if motion-reduction is preferred
+        'motion-reduce:animate-fade-in': !isDismissed,
+        'animate-fade-out': isDismissed,
+      })}
+    >
+      {children}
+    </div>
+  );
+};
+
+export type ToastMessagesProps = {
+  messages: ToastMessage[];
+  onMessageDismiss: (id: string) => void;
+  _setTimeout?: typeof setTimeout;
+};
+
+/**
+ * A collection of toast messages. These are rendered within an `aria-live`
+ * region for accessibility with screen readers.
+ */
+export default function ToastMessages({
+  messages,
+  onMessageDismiss,
+  /* istanbul ignore next - test seam */
+  _setTimeout = setTimeout,
+}: ToastMessagesProps) {
+  const [dismissedMessages, setDismissedMessages] = useState<string[]>([]);
+  const scheduledMessages = useRef(new Set<string>());
+
+  const dismissMessage = useCallback(
+    (id: string) => setDismissedMessages(ids => [...ids, id]),
+    []
+  );
+  const scheduleMessageDismiss = useCallback(
+    (id: string) => {
+      if (scheduledMessages.current.has(id)) {
+        return;
+      }
+
+      // Track that this message has been scheduled to be dismissed. After a
+      // period of time, actually dismiss it
+      scheduledMessages.current.add(id);
+      _setTimeout(() => {
+        dismissMessage(id);
+        scheduledMessages.current.delete(id);
+      }, 5000);
+    },
+    [dismissMessage, _setTimeout]
+  );
+
+  const onTransitionEnd = useCallback(
+    (direction: 'in' | 'out', message: ToastMessage) => {
+      const autoDismiss = message.autoDismiss ?? true;
+      if (direction === 'in' && autoDismiss) {
+        scheduleMessageDismiss(message.id);
+      }
+
+      if (direction === 'out') {
+        onMessageDismiss(message.id);
+        setDismissedMessages(ids => ids.filter(id => id !== message.id));
+      }
+    },
+    [scheduleMessageDismiss, onMessageDismiss]
+  );
+
+  // The `ul` containing any toast messages is absolute-positioned and the full
+  // width of the viewport. Each toast message `li` has its position and width
+  // constrained by `container` configuration in tailwind.
```

Comment:
This comment looks out of date to me. The `<ul>` is the full width of its _container_, but is not absolute positioned, and the sizing is relative to the container rather than the viewport.